### PR TITLE
User backend

### PR DESCRIPTION
**Pull request by jasmine2000**
note: need to add more middleware!
checking
- username is "valid" email
- graduation year makes sense

### CODE REVIEW

File: server/user/collection.ts [hunk 1]
```diff
@@ -18,10 +18,17 @@ class UserCollection {
...
+  static async addOne(username: string, name: string, password: string, graduationYear: string): Promise<HydratedDocument<User>> {
+    const lastLogin = new Date();
+    const nameArray = name.split(' ');
+    const user = new UserModel({
+      username,
+      password,
+      firstName: nameArray[0],
+      lastName: nameArray[-1],
+      graduationYear: parseInt(graduationYear, 10),
+      lastLogin
```
Comment (hxliu33) [inline server/user/collection.ts:30]: Change to lastActive

File: server/user/collection.ts [hunk 2]
```diff
@@ -18,10 +18,17 @@ class UserCollection {
...
    * @return {Promise<HydratedDocument<User>>} - The newly created user
    */
-  static async addOne(username: string, password: string): Promise<HydratedDocument<User>> {
-    const dateJoined = new Date();
-
-    const user = new UserModel({username, password, dateJoined});
+  static async addOne(username: string, name: string, password: string, graduationYear: string): Promise<HydratedDocument<User>> {
+    const lastLogin = new Date();
+    const nameArray = name.split(' ');
+    const user = new UserModel({
```
Comment (hxliu33) [inline server/user/collection.ts:24]: Since we’re requiring industry in UserModel we prob need to include it here right

Comment (jasmine2000) [reply server/user/collection.ts:24]: i left it out bc for whatever reason the function signatures only let me put in 4 parameters
 i figured it was the least necessary of them all and could be added later?

Comment (hxliu33) [reply server/user/collection.ts:24]: oh would the schema model validation let it pass though if it doesn't have the required parameter? we could make industry optional then ??

**Review by hxliu33 — APPROVED** [review]

**Pull request merged**